WizFi310/WizFi310.h

```diff
@@ -36,223 +36,412 @@
 #ifndef WIZFI310_H_
 #define WIZFI310_H_
 
-#include "ATCmdParser.h"
+#include "mbed.h"
+#include "mbed_trace.h"
+#include "mbed_events.h"
 #include "netsocket/WiFiAccessPoint.h"
-#include "UARTSerial.h"
+#include "RawSerial.h"
+#include "Mutex.h"
+#include "Semaphore.h"
+#include "DigitalOut.h"
+#include "Packet.h"
+
+#define WIZFI310_SOCKET_COUNT       (8)
 
 /** WizFi310Interface class.
-    This is an interface to a WizFi310Interface radio.
-*/
-class WizFi310
-{
+ *  This is an interface to a WizFi310Interface radio.
+ */
+class WizFi310 {
 public:
-    WizFi310(PinName tx, PinName rx, bool debug=false);
+    enum socket_event_t {
+        EventConnected,
+        EventDataReceived,
+        EventDataSent,
+        EventDisconnected
+    };
+    union socket_event_data_t {
+        struct socket_event_connected_t {
+            int32_t error_code;
+        } connected;
+        struct socket_event_data_received_t {
+            Packet *packet;
+        } data_received;
+        struct socket_event_data_sent_t {
+            int32_t amount_or_error;
+        } data_sent;
+        struct socket_event_disconnected_t {
 
-    /**
-    * Check firmware version of WizFi310
-    *
-    * @return character array firmware version or 0 if firmware query command gives outdated response
-    */
-    const char* get_firmware_version(void);
-    
-    /**
-    * Startup the WizFi310
-    *
-    * @param mode mode of WIFI 0-client, 1-host
-    * @return true only if WizFi310 was setup correctly
-    */
-    bool startup(int mode);
-
-    /**
-    * Reset WizFi310
-    *
-    * @return true only if WizFi310 resets successfully
-    */
-    bool reset(void);
+        } disconnected;
+    };
 
     /**
-    * Enable/Disable DHCP
-    *
-    * @param enabled DHCP enabled when true
-    * @return true only if WizFi310 enables/disables DHCP successfully
-    */
-    bool dhcp(bool enabled);
+     * Creates a new WizFir310 driver.
+     * @param tx    Tx pin name.
+     * @param rx    Rx pin name.
+     * @param rts   Rts pin name.
+     * @param cts   Cts pin name.
+     * @param rst   Rst pin name.
+     */
+    WizFi310(PinName tx, PinName rx, PinName rtx = NC, PinName cts = NC, PinName rst = NC);
```

Comment:
Should the `rtx` pin be called actually `rts`?

Reply:
Indeed, I fixed that typo. Thanks!